data/data/aws/route53/base.tf

```diff
@@ -85,11 +85,28 @@ resource "aws_route53_record" "etcd_a_nodes" {
   records = [var.etcd_ip_addresses[count.index]]
 }
 
+resource "aws_route53_record" "etcd_aaaa_nodes" {
+  count   = var.use_ipv6 == true ? var.etcd_count : 0
+  type    = "AAAA"
+  ttl     = "60"
+  zone_id = aws_route53_zone.int.zone_id
+  name    = "etcd-${count.index}.${var.cluster_domain}"
+  # TF-UPGRADE-TODO: In Terraform v0.10 and earlier, it was sometimes necessary to
+  # force an interpolation expression to be interpreted as a list by wrapping it
+  # in an extra set of list brackets. That form was supported for compatibilty in
+  # v0.11, but is no longer supported in Terraform v0.12.
+  #
+  # If the expression in the following list itself returns a list, remove the
+  # brackets to avoid interpretation as a list of lists. If the expression
+  # returns a single list item then leave it as-is and remove this TODO comment.
+  records = [var.etcd_ipv6_addresses[count.index]]
+}
+
 resource "aws_route53_record" "etcd_cluster" {
   type    = "SRV"
   ttl     = "60"
   zone_id = aws_route53_zone.int.zone_id
   name    = "_etcd-server-ssl._tcp"
-  records = formatlist("0 10 2380 %s", aws_route53_record.etcd_a_nodes.*.fqdn)
+  records = var.use_ipv6 == false ? formatlist("0 10 2380 %s", aws_route53_record.etcd_a_nodes.*.fqdn) : formatlist("0 10 2380 %s", aws_route53_record.etcd_aaaa_nodes.*.fqdn)
```

Comment:
maybe we can collect the fqdn instead of switching on entire format list?

Reply:
I don't think I understand the suggestion here.  Can you expand on it a bit?

Reply:
you are creating a formated lists in both conditionals, maybe you can condition the input list to formatlist?

Reply:
Oh, I see.  Sure, I'll try that.